onnxscript/onnx_opset/_impl/opset15.py

```diff
@@ -0,0 +1,313 @@
+        Negative axes indicate counting back from the last axis.
+        Note that axes will be clamped to the range [0, r-1], where r is the
+        rank of the input tensor if they are out-of-range (after adding r in the case of
+        negative axis). Thus, specifying any end value > r is equivalent to specifying an end
+        value of r, and specifying any start value < -r is equivalent to specifying a start
+        value of 0.
+
+        For example:
+        Input tensor with shape: [2, 3, 4]
+        No attributes specified.
+        Output: [2, 3, 4]
+
+        Input tensor with shape: [2, 3, 4]
+        start: -1
+        Output: [4]
+
+        Input tensor with shape: [2, 3, 4]
+        end: -1
+        Output: [2, 3]
+
+        Input tensor with shape: [2, 3, 4]
+        start: 1
+        end: 2
+        Output: [3]
+
+        :param data: An input tensor.
+        :param end: (Optional) Ending axis for slicing the shape. Negative value means counting dimensions from the back. If omitted, sizes of all axes upto (including) the last one will be included.
+        :param start: (Optional) Starting axis for slicing the shape. Default value is 0.Negative value means counting dimensions from the back.
+        """
+
+        schema = get_schema("Shape", 15, "")
+        op = Op(self, "Shape", schema)
+        return op(data, end=end, start=start)
+
+    def Bernoulli(
+        self, input: Union[DOUBLE, FLOAT, FLOAT16], dtype: int = None, seed: float = None
+    ) -> Union[
+        BFLOAT16,
+        BOOL,
+        DOUBLE,
+        FLOAT,
+        FLOAT16,
+        INT16,
+        INT32,
+        INT64,
+        INT8,
+        UINT16,
+        UINT32,
+        UINT64,
+        UINT8,
+    ]:
+        """
+        Draws binary random numbers (0 or 1) from a Bernoulli distribution. The input tensor should be a tensor
+        containing probabilities p (a value in the range [0,1]) to be used for drawing the binary random number,
+        where an output of 1 is produced with probability p and an output of 0 is produced with probability (1-p).
+
+        This operator is non-deterministic and may not produce the same values in different
+        implementations (even if a seed is specified).
+
+        :param input: All values in input have to be in the range:[0, 1].
+        :param dtype: The data type for the elements of the output tensor. if not specified, we will use the data type of the input tensor.
+        :param seed: (Optional) Seed to the random generator, if not specified we will auto generate one.
+        """
+
+        schema = get_schema("Bernoulli", 15, "")
+        op = Op(self, "Bernoulli", schema)
+        return op(input, dtype=dtype, seed=seed)
+
+    def BatchNormalization(
+        self,
+        X: Union[BFLOAT16, DOUBLE, FLOAT, FLOAT16],
+        scale: Union[BFLOAT16, DOUBLE, FLOAT, FLOAT16],
+        B: Union[BFLOAT16, DOUBLE, FLOAT, FLOAT16],
+        input_mean: Union[BFLOAT16, DOUBLE, FLOAT, FLOAT16],
```

Comment:
Can we capture typing constraints in ONNX that require several inputs to be of the same type using python's `TypeVar`? (Not necessarily in this PR, it could be a refinement later too.)

Reply:
I think we can refine this in a subsequent PR. The problem, AFAIK, is that Python does not support function/member level "generics". So we could generate these union types through `TypeVar`, but there will be many, and how do we name them so that they're useful in the IntelliSense context? If I am wrong, please let me know - I would love to specify generics on the functions!